python/dex/interop/jax/apply.py

```diff
@@ -245,48 +245,81 @@ def dex_call_jvp(arg_values, arg_tangents, func_atom):
   Returns:
      A pair of the primal output and the tangent.
   """
-  assert len(func_atom.compile().result_signature) == 1
+  # TODO: Make it possible to get the signature without compiling the function
+  native = get_compiled(func_atom)
+  assert len(native.result_signature) == 1
   num_args = len(arg_values)
   module = func_atom.module.copy()
 
-  # Helper functions to build strings of primal and tangent inputs.
-  def arg_string(prefix):
-    return " ".join(f"{prefix}{i}" for i in range(num_args))
+  # TODO: Support explicit arguments that are not being differentiated wrt
+  # TODO: What if some explicit primal argument occurs in the type annotation of later one?
+  # - If this happens, I have to either retain the name of the primal, or rename the annotation.
+  # - Retaining the name of the primal is a problem, though, because then a constructed tangent name
+  #   might clash with it.  (Solvable by maintaining an explicit name mapping.)
+  implicit_args = []
+  primals = []
+  tangents = []
+  name_to_ty = {}
+  for binder in native.argument_signature:
+    if binder.implicit:
+      implicit_args.append("{" + binder.name + "}")
+    else:
+      p_name = f"x{binder.name}"
+      primals.append(p_name)
+      name_to_ty[p_name] = binder.type.dex_annotation()
+      t_name = f"u{binder.name}"
```

Comment:
nit: why are tangents prefixed with u? Perhaps prefix primals with p and tangents with t?

Reply:
Because that's how it was in the old code, but sure.  Changed.